src/capwap_http_api_handler.erl

```diff
@@ -0,0 +1,378 @@
+%% Copyright (C) 2013-2017, Travelping GmbH <info@travelping.com>
+
+%% This program is free software: you can redistribute it and/or modify
+%% it under the terms of the GNU Affero General Public License as published by
+%% the Free Software Foundation, either version 3 of the License, or
+%% (at your option) any later version.
+
+%% This program is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%% GNU Affero General Public License for more details.
+
+%% You should have received a copy of the GNU Affero General Public License
+%% along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+-module(capwap_http_api_handler).
+
+-include("../include/capwap_packet.hrl").
+-include("../include/capwap_config.hrl").
+-include("../include/capwap_ac.hrl").
+
+-export([start_link/0]).
+-export([init/2, content_types_provided/2,
+         handle_request_json/2,
+         allowed_methods/2,
+         content_types_accepted/2]).
+
+-define(DEFAULT_PORT,	8000).
+-define(DEFAULT_IP,     {127, 0, 0, 1}).
+-define(ACCEPTORS_NUM,  100).
+
+-record(s, {
+    verbose = false :: boolean(),
+    opts
+}).
+
+start_link() ->
+    HttpConfig = application:get_env(capwap, http_api, []),
+    Port = proplists:get_value(port, HttpConfig, ?DEFAULT_PORT),
+    IP = proplists:get_value(ip, HttpConfig, ?DEFAULT_IP),
+    INet = get_inet(IP),
+    AcceptorsNum = proplists:get_value(acceptors_num, HttpConfig, ?ACCEPTORS_NUM),
+    Dispatch = cowboy_router:compile([
+		{'_', [
+			{"/api/v1/version",                          ?MODULE, []},
+
+			{"/api/v1/capwap/list",                      ?MODULE, []},
```

Comment:
imho the `list` part is redundant, just `/api/v1/wtp` which yields the list of the attached wtp's, `/api/v1/wtp/:id/` to get details for a concrete wtp, `/api/v1/wtp/:id/update/:link/:hash`or `/api/v1/wtp/:id/set-ssid/:ssid[/:rid]/`